packages/vue-renderer/src/renderer.js

```diff
@@ -452,11 +452,15 @@ export default class VueRenderer {
     }
   }
 
-  async renderRoute(url, context = {}) {
+  async renderRoute(url, context = {}, _retried) {
     /* istanbul ignore if */
     if (!this.isReady) {
       // Production
       if (!this.context.options.dev) {
+        if (!_retried && (this._state === 'loading' || this._state === 'created')) {
```

Comment:
Maybe add a comment to change that behavior in v3?

Reply:
```suggestion
        if (!_retried && (['loading', 'created'].includes(this._state)) {
```

Reply:
Generally uaing array causes triggering iterator (for includes) call. But as this is an error handler it is fine 😊